x-pack/plugins/enterprise_search/server/plugin.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Observable } from 'rxjs';
+import { first } from 'rxjs/operators';
+import {
+  Plugin,
+  PluginInitializerContext,
+  CoreSetup,
+  Logger,
+  SavedObjectsServiceStart,
+  IRouter,
+  KibanaRequest,
+} from 'src/core/server';
+import { UsageCollectionSetup } from 'src/plugins/usage_collection/server';
+import { UICapabilities } from 'ui/capabilities';
+import { SecurityPluginSetup } from '../../security/server';
+import { PluginSetupContract as FeaturesPluginSetup } from '../../features/server';
+
+import { checkAccess } from './lib/check_access';
+import { registerPublicUrlRoute } from './routes/enterprise_search/public_url';
+import { registerEnginesRoute } from './routes/app_search/engines';
+import { registerTelemetryRoute } from './routes/app_search/telemetry';
+import { registerTelemetryUsageCollector } from './collectors/app_search/telemetry';
+import { appSearchTelemetryType } from './saved_objects/app_search/telemetry';
+
+export interface PluginsSetup {
+  usageCollection?: UsageCollectionSetup;
+  security?: SecurityPluginSetup;
+  features: FeaturesPluginSetup;
+}
+
+export interface ServerConfigType {
+  host?: string;
+  enabled: boolean;
+  accessCheckTimeout: number;
+  accessCheckTimeoutWarning: number;
+}
```

Comment:
this seems like an unnecessary duplicate of the `ConfigType` in `x-pack/plugins/enterprise_search/server/index.ts`  ?

Reply:
Oh wow, I hadn't realized that was a reusable interface/type haha. Thanks a million 🤦‍♀️

Reply:
Addressed in 9c6acdd